src/Microsoft.SqlTools.ServiceLayer/SqlAssessment/GenerateScriptOperation.cs

```diff
@@ -0,0 +1,164 @@
+                    "SQL Assessment generate script operation failed with exception {0}",
+                    e.Message));
+
+                throw;
+            }
+            finally
+            {
+                var c = Interlocked.Exchange(ref cancellation, null);
+                c?.Dispose();
+            }
+        }
+
+        public void Cancel()
+        {
+            cancellation?.Cancel();
+        }
+
+        #region Helpers
+
+        internal static string GenerateScript(GenerateScriptParams generateScriptParams,
+            CancellationToken cancellationToken)
+        {
+            const string scriptPrologue =
+                @"IF (NOT EXISTS(SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = 'dbo' AND  TABLE_NAME = 'AssessmentResult'))
+BEGIN
+    CREATE TABLE [dbo].[AssessmentResult](
+    [CheckName] [nvarchar](max) NOT NULL,
+    [CheckId] [nvarchar](max) NOT NULL,
+    [RulesetName] [nvarchar](max) NOT NULL,
+    [RulesetVersion] [nvarchar](max) NOT NULL,
+    [Severity] [nvarchar](max) NOT NULL,
+    [Message] [nvarchar](max) NOT NULL,
+    [TargetPath] [nvarchar](max) NOT NULL,
+    [TargetType] [nvarchar](max) NOT NULL,
+    [HelpLink] [nvarchar](max) NOT NULL,
+    [Timestamp] [datetimeoffset](7) NOT NULL
+    )
+END
+GO
+INSERT INTO [dbo].[AssessmentResult] ([CheckName],[CheckId],[RulesetName],[RulesetVersion],[Severity],[Message],[TargetPath],[TargetType],[HelpLink],[Timestamp])
+VALUES";
+
+            var sb = new StringBuilder();
+            if (generateScriptParams.Items != null)
+            {
+                sb.Append(scriptPrologue);
+                foreach (var item in generateScriptParams.Items)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    if (item.Kind == AssessmentResultItemKind.Note)
+                    {
+                        sb.Append("\r\n('")
+                            .Append(Escape(item.DisplayName)).Append("','")
+                            .Append(Escape(item.CheckId)).Append("','")
+                            .Append(Escape(item.RulesetName)).Append("','")
+                            .Append(item.RulesetVersion).Append("','")
+                            .Append(item.Level).Append("','")
+                            .Append(Escape(item.Message)).Append("','")
+                            .Append(Escape(item.TargetName)).Append("','")
+                            .Append(item.TargetType).Append("','")
+                            .Append(Escape(item.HelpLink)).Append("','")
+                            .Append(
+                                item.Timestamp.ToString("yyyy-MM-dd hh:mm:ss.fff zzz", CultureInfo.InvariantCulture))
+                            .Append("'),");
+                    }
+                }
+
+                sb.Length -= 1;
+            }
+
+            return sb.ToString();
+        }
+
+        private static string Escape(string original)
```

Comment:
We already have a helper method for this [https://github.com/Microsoft/sqltoolsservice/blob/master/src/Microsoft.SqlTools.ServiceLayer/Management/Common/Utils.cs#L547](https://github.com/Microsoft/sqltoolsservice/blob/master/src/Microsoft.SqlTools.ServiceLayer/Management/Common/Utils.cs#L547)